File: flutter_client/lib/models/auth_models.dart
class LoginRequest {
  final String email;
  final String password;

  const LoginRequest({required this.email, required this.password});

  Map<String, dynamic> toJson() => {'email': email, 'password': password};
}

class UserModel {
  final int id;
  final String name;
  final String email;
  final String? role;

  const UserModel({
    required this.id,
    required this.name,
    required this.email,
    required this.role,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) => UserModel(
    id: json['id'] as int,
    name: (json['name'] ?? '') as String,
    email: (json['email'] ?? '') as String,
    role: json['role'] as String?,
  );
}

class LoginResponse {
  final String? token;
  final UserModel? user;

  const LoginResponse({required this.token, required this.user});

  factory LoginResponse.fromJson(Map<String, dynamic> json) => LoginResponse(
    token: json['token'] as String?,
    user: json['user'] is Map<String, dynamic>
        ? UserModel.fromJson(json['user'] as Map<String, dynamic>)
        : null,
  );
}

File: android-admin-app/app/src/main/java/com/alima/sinyalsahamindo/data/model/AuthModels.kt
package com.alima.sinyalsahamindo.data.model

data class LoginRequest(
    val email: String,
    val password: String
)

data class LoginResponse(
    val token: String?,
    val user: User?
)

data class User(
    val id: Int,
    val name: String,
    val email: String,
    val role: String?
)

data class TierItem(
    val id: Int,
    val name: String
)

data class TierListResponse(
    val tiers: List<TierItem> = emptyList()
)
